src/pages/Product/Product.js

```diff
@@ -0,0 +1,162 @@
+
+  useEffect(() => {
+    fetch(`/data/productData.json`)
+      .then(response => response.json())
+      .then(result => setProduct(result.data[0]));
+  }, [productId]);
+
+  // const token = localStorage.getItem('token') || ''; login창 merge되면 쓸 코드
+
+  const toBuy = e => {
+    e.preventDefault();
+    const token =
+      'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjYsImV4cCI6MTY2MTQ0MzI2NSwiaWF0IjoxNjYxMDgzMjY1fQ.KmF-Jp46fdHKwxS01SJ8PtF5yD1SkQP8rwQFA6tU9rQ';
+    const x = true;
+
+    if (x === true) {
+      fetch(`http://10.58.0.117:3000/user/cart/product/${productId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+        body: JSON.stringify({
+          quantity: count,
+        }),
+      }).then(res => {
+        if (res.ok) {
+          return navigate('/cart');
+        } else {
+          alert('로그인이 필요한 기능입니다');
+          navigate('/login');
+        }
+      });
+    }
+  };
+
+  const goToCart = e => {
+    e.preventDefault();
+    const token =
+      'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOjM3LCJleHAiOjE2NjEyOTE1MDQsImlhdCI6MTY2MTI1NTUwNH0.hMs9yhxOnyZBohpzIVqcG1T6Aej4ZU9MGCEieTUcYiQ';
+
+    const x = true;
+    if (x === true) {
+      fetch(`http://10.58.0.117:3000/user/cart/product/${productId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+        body: JSON.stringify({
+          quantity: count,
+        }),
+      }).then(res => {
+        if (res.ok) {
+          return setIsModalOpen(true);
+        } else if (res.status === 401) {
+          alert('재고');
+        } else {
+          alert('로그인이 필요한 기능입니다');
+          navigate('/login');
+        }
+      });
+    }
+  };
+
+  const countPlus = e => {
+    setCount(preCount =>
+      preCount < stock ? preCount + 1 : alert('재고 수량을 넘었습니다')
+    );
+  };
+
+  const countMinus = e => {
+    setCount(preCount => (preCount <= 0 ? 0 : (preCount = 1)));
+  };
+
+  const discountPrice = price * 0.8;
+  const priceOfInteger = Math.floor(price);
+  const priceOfTotal = priceOfInteger * count;
+
+  return (
+    <div className="product">
+      <div className="productMargin">
+        <div className="productPage">
+          <img
+            className="productImg"
+            alt="productImg"
+            src={thumbnail_image_url}
+          />
+          <form className="productForm">
+            <div className="priceTitle"> {name} </div>
+            <div className="price">
+              <div className="priceTag">{discountPrice} 원</div>
+              <div className="salePrice"> {priceOfInteger} 원 </div>
+            </div>
+            <div className="productOption">
+              <div className="optionName">{name}</div>
+              <div className="countOption">
+                <input value={count} className="countInput" />
```

Comment:
콘솔창을 보면 에러가 발생하고 있지 않나요?
input 값을 value로 수정할 때 onChange가 없으면, readOnly 처리를 합니다.
컴포넌트가 리렌더링 되면서 바껴지는 것 처럼 보여주지만, 수정할려면 기본적으로,
onChange를 함께 사용해야 합니다.
```suggestion
                <input value={count} className="countInput" onChange={(e)=>setCount(e.target.value)}/>
```

Reply:
수정 완료했습니다!